translate/lang/ro.py

```diff
@@ -0,0 +1,110 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright 2007-2009 Zuza Software Foundation
+#
+# This file is part of the Translate Toolkit.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, see <http://www.gnu.org/licenses/>.
+
+"""This module represents the Romanian language.
+
+.. seealso:: https://en.wikipedia.org/wiki/Romanian_language
+"""
+
+import re
+
+from translate.lang import common
+
+
+def ghilimele(text):
+
+    def convertquotation(match):
+        prefix = match.group(1)
+        # Let's see that we didn't perhaps match an XML tag property like
+        # <a href="something">
+        if prefix == u"=":
+            return match.group(0)
+        return u"%s„%s”" % (prefix, match.group(2))
+
+    # Check that there is an even number of double quotes, otherwise it is
+    # probably not safe to convert them.
+    if text.count(u'"') % 2 == 0:
+        text = re.sub('(.|^)"([^"]+)"', convertquotation, text)
+    singlecount = text.count(u"'")
+    if singlecount:
+        if singlecount == text.count(u'`'):
+            text = re.sub("(.|^)`([^']+)'", convertquotation, text)
+        elif singlecount % 2 == 0:
+            text = re.sub("(.|^)'([^']+)'", convertquotation, text)
+    text = re.sub(u'(.|^)“([^”]+)”', convertquotation, text)
+    return text
+
+
+class RoChecker(TranslationChecker):
+
+    @critical
+    def scedilla(self, str1, str2):
```

Comment:
These might be similar to the validchars check.

An option might be to expand that to be more generic.  I could see individual tests that could:

1. Have a list of chars that are not allowed in the target
2. Have a list of chars that should be different in the target (like your cedilla checks here)

The problem is always foreign words creating false positives.